### PR TITLE
cudaPackages: drop outdated aliases

### PR DESCRIPTION
**Pull request by srhb**
## Description of changes

Related to #306276

These traces are annoying in a lot of cases, and I think we should be good to remove them by now. I nuked all of aliases.nix for cuda-modules while I was at it. If this is heavy-handed, let me know.

For me personally, I could also live with removing this check in torch, which ends up triggering every alias in cudaPackages because of the equality check;

https://github.com/NixOS/nixpkgs/blob/04404c0900278b51a9c6e8b28b031d8b03bfb966/pkgs/development/python-modules/torch/default.nix#L206

But one or the other has to go. :smile_cat: 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by SomeoneSerge** [comment]
- 👍🏻 for removing the deprecated attributes, it's time
- 👎🏻 for removing `aliases.nix`
- How about we replace the comparison in pytorch with a comparison between `cudaPackages.cudaVersion`s? The motivation of the check was to ensure people overriding one package (magma) don't forget to override the other. For simple cases, testing `cudaVersion` would be enough, and for more complex cases well people have to take care. The package set comparison is super annoying because it needlessly forces evaluation

I'd also maybe start with the last item and check if there are any other components forcing evaluation to reach the traces.

**Comment by srhb** [comment]
@SomeoneSerge Oops, I raced you. Sure, let me do the version check instead.

**Comment by srhb** [comment]
@SomeoneSerge Alright, how's this? I had to jump some hoops to allow the empty aliases infrastructure to pass CI checks, but nothing too bad I suppose.

File: pkgs/development/python-modules/torch/default.nix [hunk 1]
```diff
@@ -203,7 +203,7 @@ let
     "MPI cudatoolkit does not match cudaPackages.cudatoolkit" =
       MPISupport && cudaSupport && (mpi.cudatoolkit != cudaPackages.cudatoolkit);
     "Magma cudaPackages does not match cudaPackages" =
-      cudaSupport && (effectiveMagma.cudaPackages != cudaPackages);
+      cudaSupport && (effectiveMagma.cudaPackages.cudaVersion != cudaPackages.cudaVersion);
```
Comment (SomeoneSerge) [inline pkgs/development/python-modules/torch/default.nix:206]:
```suggestion
    # This used to be a deep package set comparison, but it made it evaluation too strict. This was bad e.g. for cudaPackages' `aliases.nix`.
    "Magma cudaPackages does not match cudaPackages. To apply overrides consistently, use overlays" =
      cudaSupport && (effectiveMagma.cudaPackages.cudaVersion != cudaPackages.cudaVersion);
```

File: pkgs/development/cuda-modules/aliases.nix
Comment (SomeoneSerge) [inline pkgs/development/cuda-modules/aliases.nix:1]: ofborg asking for nixfmt

File: pkgs/development/cuda-modules/aliases.nix [hunk 2]
```diff
@@ -1,18 +1,13 @@
...
-  autoFixElfFiles = mkRenamed "autoFixElfFiles" "pkgs.autoFixElfFiles" final.pkgs.autoFixElfFiles; # Added 2024-03-30
-  autoAddDriverRunpath =
-    mkRenamed "autoAddDriverRunpath" "pkgs.autoAddDriverRunpath"
-      final.pkgs.autoAddDriverRunpath; # Added 2024-03-30
-  autoAddOpenGLRunpathHook =
-    mkRenamed "autoAddOpenGLRunpathHook" "pkgs.autoAddDriverRunpathHook"
-      final.pkgs.autoAddDriverRunpath; # Added 2024-03-30
-}
+  # mkRenamed =
+  #   oldName: newName: newPkg:
```
Comment (SomeoneSerge) [inline pkgs/development/cuda-modules/aliases.nix:10]: An arbitrary alternative: one could keep `mkRenamed` and return a `mapAttrs mkRenamed { }`. I'm fine with the comment though

File: pkgs/development/python-modules/torch/default.nix (hunk 1 above)
Comment (srhb) [reply pkgs/development/python-modules/torch/default.nix:212]: Done this, but tweaked the wording a bit.

File: pkgs/development/cuda-modules/aliases.nix (hunk 2 above)
Comment (srhb) [reply pkgs/development/cuda-modules/aliases.nix]: Don't want to do this, not arbitrary

File: pkgs/development/python-modules/torch/default.nix [hunk 3]
```diff
@@ -202,8 +202,14 @@ let
...
       MPISupport && cudaSupport && (mpi.cudatoolkit != cudaPackages.cudatoolkit);
+    # This used to be a deep package set comparison between cudaPackages and
+    # effectiveMagma.cudaPackages, but it made it evaluation too strict. This
+    # was bad e.g. for cudaPackages' `aliases.nix`, forcing warnings upon
+    # everyone using torch even if the alias was never used.
+    "Magma cudaPackages does not match cudaPackages. To apply overrides consistently, use overlays" =
+      cudaSupport && (effectiveMagma.cudaPackages.cudaVersion != cudaPackages.cudaVersion);
     "Magma cudaPackages does not match cudaPackages" =
-      cudaSupport && (effectiveMagma.cudaPackages != cudaPackages);
+      cudaSupport && (effectiveMagma.cudaPackages.cudaVersion != cudaPackages.cudaVersion);
```
Comment (SomeoneSerge) [inline pkgs/development/python-modules/torch/default.nix:212]: I think we got two checks now:)

Comment (srhb) [reply pkgs/development/python-modules/torch/default.nix:210]: :scream_cat: Sorry. Me trying to do two things at the same time. I think it's okay now, if I didn't anger nixfmt again.

File: pkgs/development/cuda-modules/aliases.nix [hunk 4]
```diff
@@ -1,18 +1,13 @@
...
-in
+# There are currently no aliases , but the infrastructure is desired preserved.
+# To please the nixpkgs CI check for nixf-tidy, we must not have unused
+# definitions here. mkRenamed is thus commented out until it is needed again.
+#
+# let
+#   mkRenamed =
+#     oldName: newName: newPkg:
+#     final.lib.warn "cudaPackages.${oldName} is deprecated, use ${newName} instead" newPkg;
+# in
```
Comment (SomeoneSerge) [inline pkgs/development/cuda-modules/aliases.nix:11]: Ah, here we go: https://github.com/NixOS/nixpkgs/issues/331085

File: pkgs/development/python-modules/torch/default.nix [hunk 5]
```diff
@@ -202,8 +202,12 @@ let
       ]);
     "MPI cudatoolkit does not match cudaPackages.cudatoolkit" =
       MPISupport && cudaSupport && (mpi.cudatoolkit != cudaPackages.cudatoolkit);
+    # This used to be a deep package set comparison between cudaPackages and
+    # effectiveMagma.cudaPackages, but it made it evaluation too strict. This
```
Comment (SomeoneSerge) [inline pkgs/development/python-modules/torch/default.nix:206]: made it/evaluation

Comment (SomeoneSerge) [reply pkgs/development/python-modules/torch/default.nix]: I pushed a fix to avoid the extra roundtrip:)

**Pull request merged**